File: java/allergies/src/main/java/Allergies.java
import java.util.List;
import java.util.ArrayList;

public class Allergies {
	
	private final int score;
	
	Allergies(int score){
		this.score = score;
	}
	
	boolean isAllergicTo(Allergen allergen) {
		return getList().contains(allergen);
	}
	
	List<Allergen> getList(){
		List<Allergen> allergies = new ArrayList<>(); 
		for (Allergen allergen : Allergen.values())
			if ((score & allergen.getScore()) > 0)
				allergies.add(allergen);
		return allergies;
	}
	
}

File: python/allergies/allergies.py
class Allergies(object):

    def __init__(self, score):
        self.score = [allergen for num, allergen in list(enumerate([
            'eggs',
            'peanuts',
            'shellfish',
            'strawberries',
            'tomatoes',
            'chocolate',
            'pollen',
            'cats'
            ]))
            if 0 < (score & (1 << num))]

    def is_allergic_to(self, item):
        return item in self.score

    @property
    def lst(self):
        return self.score
